Sources/TuistAcceptanceTesting/FixtureTestingTrait.swift

```diff
@@ -0,0 +1,103 @@
+                        let projectHandle = String(UUID().uuidString.prefix(12).lowercased())
+                        let fullHandle = "\(organizationHandle)/\(projectHandle)"
+                        let email = try #require(ProcessInfo.processInfo.environment[EnvKey.authEmail.rawValue])
+                        let password = try #require(ProcessInfo.processInfo.environment[EnvKey.authPassword.rawValue])
+
+                        try await TuistTest.run(
+                            LoginCommand.self,
+                            ["--email", email, "--password", password, "--path", fixtureTemporaryDirectory.pathString]
+                        )
+                        try await TuistTest.run(
+                            OrganizationCreateCommand.self,
+                            [organizationHandle, "--path", fixtureTemporaryDirectory.pathString]
+                        )
+                        try await TuistTest.run(
+                            ProjectCreateCommand.self,
+                            [fullHandle, "--path", fixtureTemporaryDirectory.pathString]
+                        )
+
+                        try await FileSystem().writeText("""
+                        import ProjectDescription
+
+                        let config = Config(
+                            fullHandle: "\(fullHandle)",
+                            url: "\(Environment.current.variables["TUIST_URL"] ?? "https://canary.tuist.dev")"
+                        )
+                        """, at: fixtureTemporaryDirectory.appending(components: "Tuist.swift"), options: Set([.overwrite]))
+                        resetUI()
+
+                        let revert = {
+                            try await TuistTest.run(
+                                ProjectDeleteCommand.self,
+                                [fullHandle, "--path", fixtureTemporaryDirectory.pathString]
+                            )
+                            try await TuistTest.run(
+                                OrganizationDeleteCommand.self,
+                                [organizationHandle, "--path", fixtureTemporaryDirectory.pathString]
+                            )
+                            try await TuistTest.run(
+                                LogoutCommand.self
+                            )
+                        }
+
+                        do {
+                            try await function()
+                        } catch {
+                            try await revert()
+                            throw error
+                        }
+                        try await revert()
+                    }
+                }
+        }
+    }
+}
+
+extension Trait where Self == FixtureTestingTrait {
+    public static func withFixtureConnectedToCanary(
+        _ fixture: String,
+        fixturesDirectory: AbsolutePath = FixtureTestingTrait.defaultFixturesDirectory()
```

Comment:
I wonder if we can get rid of this pattern since we don't really have that many fixtures in the `cli-ext` and those we do, should be relatively easy to move in `tuist/tuist`, which is the better long-term setup.